x-pack/plugins/cases/server/services/user_actions/operations/create.ts

```diff
@@ -399,15 +395,56 @@ export class UserActionPersister {
         payload,
       });
 
-      if (userAction) {
-        await this.createAndLog({ userAction, refresh });
+      if (userActionPayload) {
+        await this.createAndLog({ userAction: userActionPayload, refresh });
       }
     } catch (error) {
       this.context.log.error(`Error on creating user action of type: ${type}. Error: ${error}`);
       throw error;
     }
   }
 
+  public async bulkCreateUserAction<T extends keyof BuilderParameters>({
+    userActions,
+    refresh,
+  }: BulkCreateUserActionArgs<T>): Promise<void> {
+    try {
+      this.context.log.debug(`Attempting to bulk create a user actions`);
+
+      if (userActions.length <= 0) {
+        return;
+      }
+
+      const userActionsPayload = userActions
+        .map((ua) => {
+          const { action, type, caseId, user, owner, payload, connectorId, attachmentId } = ua;
```

Comment:
nit: might as well do
```suggestion
        .map(({ action, type, caseId, user, owner, payload, connectorId, attachmentId }) => {
```